### PR TITLE
Subscription cleanup

### PR DESCRIPTION
**Pull request by fil512**
Fix a couple of small issues with subscription cleanup

### CODE REVIEW

File: hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/match/registry/SubscriptionLoader.java [hunk 1]
```diff
@@ -209,22 +206,39 @@ private int updateSubscriptionRegistry(List<IBaseResource> theResourceList) {
...
+		// internally, subscriptions that cannot activate
+		// will be set to error
+		boolean activated = mySubscriptionActivatingInterceptor.activateSubscriptionIfRequired(theSubscription);
+		if (activated) {
+			return true;
+		}
+		logSubscriptionNotActivatedPlusErrorIfPossible(theSubscription);
+		return false;
+	}
+
```
Comment (VK-SMILECDR) [inline hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/match/registry/SubscriptionLoader.java:227]:
```
	/**
	 * @param theSubscription
	 * @return true if activated
	 */
	private boolean activateSubscriptionIfRequested(IBaseResource theSubscription) {
		if (SubscriptionConstants.REQUESTED_STATUS.equals(mySubscriptionCanonicalizer.getSubscriptionStatus(theSubscription))) {
			// internally, subscriptions that cannot activate will be set to error
			if (mySubscriptionActivatingInterceptor.activateSubscriptionIfRequired(theSubscription)) {
				return true;
			}
			logSubscriptionNotActivatedPlusErrorIfPossible(theSubscription);
		}
		return false;
	}
```

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/messaging/BaseResourceMessage.java [hunk 2]
```diff
@@ -176,6 +177,24 @@ public void setMessageKey(String theMessageKey) {
...
+				default:
+				case UPDATE:
+					operationType = RestOperationTypeEnum.UPDATE;
+					break;
+				case DELETE:
+					operationType = RestOperationTypeEnum.DELETE;
+					break;
+			}
+			return operationType;
+		}
```
Comment (VK-SMILECDR) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/messaging/BaseResourceMessage.java:198]: Very confusing code....  Where is a test?

```
	public enum OperationTypeEnum {
		CREATE(RestOperationTypeEnum.CREATE),
		UPDATE(RestOperationTypeEnum.UPDATE),
		DELETE(RestOperationTypeEnum.DELETE),
		MANUALLY_TRIGGERED(null),
		TRANSACTION(null);

		private RestOperationTypeEnum myRestOperationTypeEnum;

		OperationTypeEnum(RestOperationTypeEnum theRestOperationTypeEnum) {
			myRestOperationTypeEnum = theRestOperationTypeEnum;
		}

		public RestOperationTypeEnum asRestOperationType() {
			return myRestOperationTypeEnum;
		}
	}

```

**Review by VK-SMILECDR — CHANGES_REQUESTED** [review]

File: hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/match/registry/SubscriptionLoader.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/match/registry/SubscriptionLoader.java:227]: Much cleaner!

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/messaging/BaseResourceMessage.java (hunk 2 above)
Comment (fil512) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/messaging/BaseResourceMessage.java:198]: test is in cdr.  it's just a convenience enum converter that used to be outside the enum.  I find converters are more helpful when they're in the source object.

**Review by VK-SMILECDR — APPROVED** [review]

**Pull request merged**